### ISSUE TITLE
MSVC Compiler warnings in Helpers.hh

### ISSUE THREAD

**Issue by scpeters** [gazebosim/gz-math#527]
## Environment
* OS Version: Windows 10
* Source or binary build? source build of ign-math6

## Description
* Expected behavior: it should compiler without compiler warnings
* Actual behavior: it has multiple compiler warnings about [Helpers.hh line 254](https://github.com/gazebosim/gz-math/blob/ignition-math6_6.13.0/include/gz/math/Helpers.hh#L954)

~~~
C:\J\workspace\ign_math-ign-6-win\ws\gz-math\include\gz/math/Helpers.hh(954,20):
 warning C4834: discarding return value of function with 'nodiscard' attribute
~~~

## Steps to reproduce

1. [Run a CI build](https://build.osrfoundation.org/view/ign-citadel/job/ign_math-ign-6-win/build?delay=0sec) on one of the newer windows hosts (such as [win-ec2.nv.WinS22-9e6bbc45](https://build.osrfoundation.org/computer/win-ec2.nv.WinS22-9e6bbc45))
2. 
3.

### Older MSVC version doesn't have compiler warnings

~~~
-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19044.
-- The C compiler identification is MSVC 19.29.30143.0
-- The CXX compiler identification is MSVC 19.29.30143.0
~~~

[![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=ign_math-ign-6-win&build=67)](https://build.osrfoundation.org/view/ign-citadel/job/ign_math-ign-6-win/67/) https://build.osrfoundation.org/view/ign-citadel/job/ign_math-ign-6-win/67/

### Newer MSVC version has compiler warnings

~~~
-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.20348.
-- The C compiler identification is MSVC 19.33.31630.0
-- The CXX compiler identification is MSVC 19.33.31630.0
~~~

[![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=ign_math-ign-6-win&build=66)](https://build.osrfoundation.org/view/ign-citadel/job/ign_math-ign-6-win/66/) https://build.osrfoundation.org/view/ign-citadel/job/ign_math-ign-6-win/66/

~~~
C:\J\workspace\ign_math-ign-6-win\ws\gz-math\include\gz/math/Helpers.hh(954,20):
 warning C4834: discarding return value of function with 'nodiscard' attribute
 (compiling source file C:\J\workspace\ign_math-ign-6-win\ws\gz-math\src\Color.cc)
 [C:\J\workspace\ign_math-ign-6-win\ws\build\ignition-math6\src\ignition-math6.vcxproj]
~~~

**Comment by sdhar04** [comment]
@azeey I would like to work on this issue. However, the links under Steps To Reproduce aren't working. Could you give me some pointers on how to tackle this? Should I remove the attribute 'nodiscard' of the given function?

**Comment by azeey** [comment]
Thanks for your interest @sdhar04. In trying to reproduce this myself, I found out that the issue has been resolved in #564. So, I'll go ahead and close it. Please feel free to find another "good first issue".
